src/CodeGenerator/MethodArgument.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CodeGenerator
+{
+    public class MethodArgument
+    {
+        private static readonly string[] AllowedModifiers =
```

Comment:
I suppose I should add an enum for this to be consistent with `ClassAccessibilityLevel`/`MethodAccessibilityLevel`.